src/Profiles/Fec/DataPages/ANTPLUS_FecBaseMainDataPage.cpp

```diff
@@ -0,0 +1,72 @@
+#include <Profiles/Fec/DataPages/ANTPLUS_FecBaseMainDataPage.h>
+#include <Profiles/Fec/ANTPLUS_FecPrivateDefines.h>
+#include <ANTPLUS_PrivateDefines.h>
+
+#define DATAPAGE_BYTE 0
+#define TOGGLE_BYTE   0
+#define HEARTBEATEVENTTIME_LSB_BYTE 4
+#define HEARTBEATEVENTTIME_MSB_BYTE 5
+#define HEARTBEATCOUNT_BYTE 6
+#define COMPUTEDHEARTRATE_BYTE 7
+#define DATAPAGE_MASK 0x7F
+#define TOGGLE_MASK   0x80
+#define TOGGLE_SHIFT  7
+
+template<class T>
+FecCoreMainDataPage<T>::FecCoreMainDataPage() : CoreDataPage<T>() {}
+
+template<class T>
+uint8_t FecCoreMainDataPage<T>::getDataPageNumber() {
+    return this->get8BitValue(DATAPAGE_BYTE, DATAPAGE_MASK);
+}
+
+template<class T>
+uint8_t FecCoreMainDataPage<T>::getPageChangeToggle() {
+    return this->get8BitValue(TOGGLE_BYTE, TOGGLE_MASK, TOGGLE_SHIFT);
+}
+
+template<class T>
+uint16_t FecCoreMainDataPage<T>::getHeartBeatEventTime() {
```

Comment:
I should really document this, but the "core" datapages are meant to the base and handle any common functionality. For example. HR uses custom datapages where all the fields are the same except 3 bytes so a bunch of the fields are in there. I don't think FEC has these so these should probably be removed. The type can stay if you see value in using it to get the page number.